rust-lisa/src/main/java/it/unipr/cfg/expression/numeric/RustDivExpression.java

```diff
@@ -46,8 +48,17 @@ T extends TypeDomain<T>> AnalysisState<A, H, V, T> binarySemantics(
 					InterproceduralAnalysis<A, H, V, T> interprocedural, AnalysisState<A, H, V, T> state,
 					SymbolicExpression left, SymbolicExpression right, StatementStore<A, H, V, T> expressions)
 					throws SemanticException {
-		// TODO too coarse
-		return state.top();
+
+		AnalysisState<A, H, V, T> result = state.bottom();
+
+		for (Type leftType : left.getRuntimeTypes())
+			for (Type rightType : right.getRuntimeTypes())
+				if (leftType.canBeAssignedTo(rightType) && rightType.canBeAssignedTo(leftType))
```

Comment:
We should also check that left and right types are numeric types.

Reply:
It is possibile to have operator overload in rust, so it is not guaranteed that those types will be numerical. As agreed in a previous conversation, we should revise those function when we will be modelling operator overload.